components/security-mgt/org.wso2.carbon.security.mgt/src/main/java/org/wso2/carbon/security/keystore/KeyStoreManagementServiceImpl.java

```diff
@@ -65,6 +96,12 @@
  */
 public class KeyStoreManagementServiceImpl implements KeyStoreManagementService {
 
+    private static final Log log = LogFactory.getLog(KeyStoreManagementServiceImpl.class);
+    private static final String CERT_PEM_START = "-----(BEGIN CERTIFICATE)-----";
+    private static final String CERT_PEM_END = "-----(END CERTIFICATE)-----";
+    private static final Pattern PATTERN = Pattern.compile(CERT_PEM_START + "([^-]+)" + CERT_PEM_END);
```

Comment:
Can we do the pattern compile at the class construct phase or better yet at the component activation, not at the class load phase. That way the server startup delay caused by this can be avoided.